bindings/rust/s2n-tls/src/raw/connection.rs

```diff
@@ -170,6 +172,21 @@ impl Connection {
         Ok(self)
     }
 
+    pub(crate) fn config(&self) -> Result<Option<Config>, Error> {
+        let mut raw = core::ptr::null_mut();
+        let config = unsafe {
+            s2n_connection_get_config(self.connection.as_ptr(), &mut raw)
+                .into_result()
+                .map_or_else(|_err| None, |_ok| Some(raw))
+                .and_then(|raw| NonNull::new(raw).map(|raw| Config::from_raw(raw)))
+        };
+        // Because the config pointer is still set on the connection, this is a copy,
+        // not the original config. This is fine -- Configs are immutable.
+        let clone = config.clone();
```

Comment:
`config` can be None here which means dont actually have a handle to a config. This fn could be simplified by just returning `Result<Config, Error>`. 

https://github.com/aws/s2n-tls/blob/main/tls/s2n_connection.c#L1545-L1547